plugins/sphinx_search/lib/SphinxCategoryCriteria.php

```diff
@@ -143,8 +143,8 @@ protected function applyFilterFields(baseObjectFilter $filter)
 		{
 			$names = $filter->get('_likex_name_or_reference_id');
 			KalturaLog::debug("Attach free text [$names]");
-			
-			$this->addFreeTextToMatchClauseByMatchFields($names, categoryFilter::NAME_REFERNCE_ID, null, true);
+			$additional = array("(@(name,reference_id) $names)");
```

Comment:
Please move the implementation into addFreeTextToMatchClauseByMatchFields and check also for KalturaUserFilter::idOrScreenNameStartsWith and firstNameOrLastNameStartsWith.